### PR TITLE
Disable Warlords Damage spam

### PR DESCRIPTION
**Pull request by Michaelmvv**
Add config options to disable warlords spam.

### CODE REVIEW

File: src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java [hunk 1]
```diff
@@ -73,15 +75,15 @@
     public static final String PAINTBALL_KILLSTREAK_TRACKER = "Show/Hide your active Paintball killstreaks and the ones on cooldown.";
     public static final String KILLSTREAK_TRACKER = "Enable/Disable tracking how many kills you can get within one live in ";
     public static final String BLITZ_DEATHMATCH_NOTIFIER = "Show/Hide a little flashing message when death match is near in Blitz Survival Games";
-    public static final String WITHER_COIN_COUNTER  = "Turn on/off the Counting of the Coins for Wither Damage in Mega Walls.";
+    public static final String WITHER_COIN_COUNTER = "Turn on/off the Counting of the Coins for Wither Damage in Mega Walls.";
 
     public HudPixelConfig(File config) {
         // the "3" is the defined config version
-        this.config = new Configuration(config, "3");
+        this.config = new Configuration(config, "4");
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java]: Why? This is not necessary for adding new settings. It is only needed for renaming and updating existing settings.

File: src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java [hunk 2]
```diff
@@ -137,13 +141,14 @@ public Configuration getConfigFile() {
         return config;
     }
 
-    // for the config gui introduced in 2.1.0, most properties need to be renamed
+    // for the config gui introduced in 2.1.0, most properties need to be
+    // renamed
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java]: Did you auto-format this file? It's probably a setting to break comments after 80 chars. However my code style doesn't follow this. For this reason, I'll revert the comments when I merge this.

File: src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java [hunk 3]
```diff
@@ -157,34 +162,38 @@ private void updateConfig(String oldVersion) {
...
+            // for 2.0.0 as it was disabled
         }
-        if(fallThrough || oldVersion.equals("2")) {
+        if (fallThrough || oldVersion.equals("2")) {
             fallThrough = true;
-            
-            this.config.get(DISPLAY_CATEGORY, "displayTipAllButton",false, "Show a button that runs /tip all.").set(false);;
+
+            this.config.get(DISPLAY_CATEGORY, "displayTipAllButton", false, "Show a button that runs /tip all.").set(false);
+            ;
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java]: ...

File: src/main/java/com/palechip/hudpixelmod/HudPixelConfigGui.java [hunk 4]
```diff
@@ -22,10 +23,12 @@
         // these categories have to be there always
         list.addAll(new ConfigElement(configFile.getCategory(Configuration.CATEGORY_GENERAL)).getChildElements());
         list.addAll(new ConfigElement(configFile.getCategory(HudPixelConfig.DISPLAY_CATEGORY)).getChildElements());
-        if(HudPixelMod.instance().gameDetector.getCurrentGame() != null) {
+        if (HudPixelMod.instance().gameDetector.getCurrentGame() != null) {
             // Add the current game
             list.addAll(new ConfigElement(configFile.getCategory(HudPixelMod.instance().gameDetector.getCurrentGame().getConfigCategory())).getChildElements());
-            new ChatMessageComposer("Opened the config for currently played game. Open in lobby to access all options.", EnumChatFormatting.DARK_GREEN).send();
+            if (HypixelNetworkDetector.isHypixelNetwork) {
+                new ChatMessageComposer("Opened the config for currently played game. Open in lobby to access all options.", EnumChatFormatting.DARK_GREEN).send();
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/HudPixelConfigGui.java]: This crashed the game, didn't it? I noticed it when developing but I forgot to fix it. Thanks.

File: src/main/java/com/palechip/hudpixelmod/HudPixelMod.java [hunk 5]
```diff
@@ -1,3 +1,10 @@
+
+
+
+
+
+
+
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/HudPixelMod.java]: ?

File: src/main/java/com/palechip/hudpixelmod/chat/WarlordsDamageChatDisable.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.palechip.hudpixelmod.chat;
+
+import com.palechip.hudpixelmod.HudPixelConfig;
+
+import net.minecraftforge.client.event.ClientChatReceivedEvent;
+
+public class WarlordsDamageChatDisable {
+    public static final String takeDamage = "�";
+    public static final String giveDamage = "�";
```
Comment (palechip) [inline src/main/java/com/palechip/hudpixelmod/chat/WarlordsDamageChatDisable.java]: Please please test a build of the mod. I'm fairly certain that Java will screw up the `»` an the `«` causing the feature not to work. A similar thing happened to me with the color code sign in the past. You'll probably have to write it using \u...

**Comment by palechip** [comment]
When you opened the issue suggesting this change, I started thinking about how to improve the Warlords chat. My idea was to bundle the chat messages and show how much damage you've done every second or two. Please tell me what you think about it.

Anyway thanks for taking action yourself!

File: src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java (hunk 1 above)
Comment (Michaelmvv) [inline src/main/java/com/palechip/hudpixelmod/HudPixelConfig.java]: Sorry. I thought I changed that back to 3.

File: src/main/java/com/palechip/hudpixelmod/chat/WarlordsDamageChatDisable.java (hunk 6 above)
Comment (Michaelmvv) [inline src/main/java/com/palechip/hudpixelmod/chat/WarlordsDamageChatDisable.java]: I tested this on my computer and it worked.

**Comment by Michaelmvv** [comment]
I didn't even think of that approach. I think that is a great idea, and many people would like it, but personally I don't want any damage/health messages in chat.

**Pull request merged**